src/lib.rs

```diff
@@ -750,6 +760,45 @@ impl<T> Drop for Sender<T> {
     }
 }
 
+/// A sender that does not prevent the channel from being closed.
+///
+/// Weak senders do not count towards the number of active senders on the channel. As soon as
+/// all normal [`Sender`]s are dropped, the channel is closed, even if there is still a
+/// `WeakSender`.
+///
+/// To send messages, a `WeakSender` must first be upgraded to a `Sender` using the [`upgrade`]
+/// method.
+pub struct WeakSender<T> {
+    shared: Weak<Shared<T>>,
+}
+
+impl<T> WeakSender<T> {
+    /// Tries to upgrade the `WeakSender` to a [`Sender`], in order to send messages.
+    ///
+    /// Returns `None` if the channel was closed already. Note that a `Some` return value
+    /// does not guarantee that the channel is still open.
+    pub fn upgrade(&self) -> Option<Sender<T>> {
```

Comment:
The convention so far has been to return `Result<..., ...>` for functions akin to this. I realise that `Option` is just as expressive in this case, but would you be willing to change it for the sake of consistency with `std::sync::mpsc`'s API?

Reply:
I chose `Option` because that's what the comparable methods of `tokio` ([`WeakSender::upgrade`](https://docs.rs/tokio/1.21.2/tokio/sync/mpsc/struct.WeakSender.html#method.upgrade)) and the standard library ([`Weak::upgrade`](https://doc.rust-lang.org/stable/std/sync/struct.Weak.html#method.upgrade)) use. I'm of course happy to change it to a `Result` if you prefer.


> for the sake of consistency with `std::sync::mpsc`'s API?

Could you clarify which part of the API you mean?